src/v1/device_manager_client.ts

```diff
@@ -1532,23 +1584,73 @@ export class DeviceManagerClient {
       {} | undefined
     ]
   > | void {
-    request = request || {};
-    let options: CallOptions;
-    if (typeof optionsOrCallback === 'function' && callback === undefined) {
-      callback = optionsOrCallback;
-      options = {};
-    } else {
-      options = optionsOrCallback as CallOptions;
-    }
-    options = options || {};
-    options.otherArgs = options.otherArgs || {};
-    options.otherArgs.headers = options.otherArgs.headers || {};
-    options.otherArgs.headers['x-goog-request-params'] =
-      this._gaxModule.routingHeader.fromParams({
-        name: request.name || '',
+    return new Promise(async (resolve, reject) => {
+      const token_response = await this.getRegistryToken();
+      const token = JSON.parse(token_response);
+      const payload = JSON.stringify({
+        name: request?.name,
       });
-    this.initialize();
-    return this.innerApiCalls.listDeviceStates(request, options, callback);
+      var options = {
+        host: 'iot-sandbox.clearblade.com',
+        path: `/api/v/1/code/` + token.systemKey + `/devicesStatesList`,
```

Comment:
Please use the webhook endpoint instead. @prashantdatirIngress could you please share the webhook information with @ronak-ingress ?

Reply:
![image](https://user-images.githubusercontent.com/115336432/197265174-2db0b614-3e46-4e16-9cdf-4742fb3d009f.png)

Please refer the postman collection shared by Akash
@ronak-ingress